prow/plugins/releasenote/releasenote_test.go

```diff
@@ -633,3 +634,121 @@ func TestShouldHandlePR(t *testing.T) {
+				Action: github.IssueCommentActionCreated,
+				Issue:  github.Issue{Number: issueNum, User: github.User{Login: "user"}},
+				Comment: github.IssueComment{
+					Body: "/release-note-edit\r\nNew note",
+					User: github.User{Login: "user"},
+				},
+				Repo: github.Repo{Owner: github.User{Login: "org"}},
+			},
+			comment: "release note block",
+			fcFunc: func(fc *fakegithub.FakeClient) {
+				fc.OrgMembers["org"] = []string{"user"}
+			},
+		},
+		{
+			name: "multiple release note blocks",
+			event: github.IssueCommentEvent{
+				Action: github.IssueCommentActionCreated,
+				Issue:  github.Issue{Number: issueNum, User: github.User{Login: "user"}},
+				Comment: github.IssueComment{
+					Body: "/release-note-edit\r\n```release-note\r\nThe new note\r\n```\r\n```release-note\r\nThe second note\r\n```\r\n",
+					User: github.User{Login: "user"},
+				},
+				Repo: github.Repo{Owner: github.User{Login: "org"}},
+			},
+			comment: "single release note block",
+			fcFunc: func(fc *fakegithub.FakeClient) {
+				fc.OrgMembers["org"] = []string{"user"}
+			},
+		},
+		{
+			name: "happy path",
+			event: github.IssueCommentEvent{
+				Action: github.IssueCommentActionCreated,
+				Issue:  github.Issue{Number: issueNum, User: github.User{Login: "user"}, Body: "Top level\r\n```release-note\r\nNONE\r\n```\r\n"},
+				Comment: github.IssueComment{
+					Body: "/release-note-edit\r\n```release-note\r\nThe new note\r\n```\r\n",
+					User: github.User{Login: "user"},
+				},
+				Repo: github.Repo{Owner: github.User{Login: "org"}},
+			},
+			fcFunc: func(fc *fakegithub.FakeClient) {
+				fc.OrgMembers["org"] = []string{"user"}
+				fc.Issues[issueNum] = &github.Issue{
+					Number: issueNum,
+					User:   github.User{Login: "user"},
+					Body:   "Top level\r\n```release-note\r\nNONE\r\n```\r\n",
+				}
+			},
+			expectEdit: true,
+		},
+	}
+	for _, tc := range ts {
+		t.Run(tc.name, func(t *testing.T) {
+			fc := fakegithub.NewFakeClient()
+			if tc.fcFunc != nil {
+				tc.fcFunc(fc)
+			}
+			err := editReleaseNote(fc, logrus.WithField("plugin", PluginName), tc.event)
+			if err != nil {
+				if !tc.expectError {
+					t.Fatalf("unexpected error: %v", err)
+				}
+				if m := err.Error(); !strings.Contains(m, tc.errorMessage) {
+					t.Fatalf("expected error to contain: %s got: %v", tc.errorMessage, m)
+				}
+			}
+			if err == nil && tc.expectError {
+				t.Fatalf("expected error but did not produce")
+			}
+			if len(tc.comment) != 0 {
+				if cm, ok := fc.IssueComments[tc.event.Issue.Number]; ok {
+					if !strings.Contains(cm[0].Body, tc.comment) {
+						t.Fatalf("expected comment to contain: %s got: %s", tc.comment, cm[0].Body)
+					}
+				}
+			}
+			if len(tc.comment) == 0 && len(fc.IssueComments[issueNum]) != 0 {
+				t.Fatalf("unexpected comment: %v", fc.IssueComments[issueNum])
+			}
+			if tc.expectEdit {
+				if !strings.Contains(fc.Issues[issueNum].Body, "```release-note\r\nThe new note\r\n```") {
```

Comment:
Can we compare the full body string here instead of a substring check? The current check would succeed even if we were to just append the new block to the existing body